micrometer-core/src/main/java/io/micrometer/core/aop/CountedAspect.java

```diff
@@ -180,7 +180,7 @@ private Function<ProceedingJoinPoint, Iterable<Tag>> makeSafe(
                 return function.apply(pjp);
             }
             catch (Throwable t) {
-                joinPointTagsFunctionLogger
+                WARN_THEN_DEBUG_LOGGER
```

Comment:
I wonder if we want the two catch cases to share the same `WarnThenDebugLogger`. With them shared, if an application is throwing exceptions from both places only whichever one runs first will be logged at warn, masking the other. Separate loggers would ensure they both get logged at warn the first time.

Reply:
I don't have a strong opinion either. We can leave it up to @jonatan-ivanov to decide :)

Reply:
[pressure intensifies...] I flipped a coin, it's tails! Oh wait, I guess we did not agree what happens when it's tails. 🙂

I like the idea but I think I would probably leave it as-is (two loggers) for now because:
- (It's tails! 🙂)
- It's easier to discover the source of the log event with the fully qualified class name (unlikely but the stacktrace can be missing)
- It can alert the users about the issue with `@Timed` and `@Counted` too in one run so they can fix both and don't need to restart/redeploy the app
- Users can configure different behaviors for the two loggers (unlikely that they will)
- It does not add much overhead (one logger and one extra WARN event)